### PR TITLE
Fix documentation for GitHub enterprise

### PR DESCRIPTION
**Pull request by byroot**
Fixes: https://github.com/Shopify/shipit-engine/issues/430

@mathieugagne is this  good for you?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -283,7 +283,7 @@ development:
 
 If you specify an `access_token`, you don't need a `login` and `password`. The opposite is also true:  if you specify a `login` and `password`, then you don't need an `access_token`.
 
-The `api_endpoint` is optional, and required only if you want to use a proxy.
```
Comment (gmalette) [inline README.md]: `If you use GitHub Enterprise or a proxy` ?

Comment (byroot) [inline README.md]: I don't think the proxy use case is common enough to be worth a special case.

Comment (gmalette) [inline README.md]: I guess, yeah, it's a lot of trouble

**Comment by mathieugagne** [comment]
:ok_hand: I looked at the doc but I guessed I must have skipped this line. Thanks for the prompt response guys.

**Pull request merged**